tcms/testplans/tests/test_views.py

```diff
@@ -54,3 +55,127 @@ def test_view_tags_without_permissions(self):
         # assert tag actions are shown
         self.assertNotContains(response, 'Add Tag')
         self.assertContains(response, '<span class="disabled grey">%s</span>' % _('Remove'))
+
+
+class EditPlanViewTest(LoggedInTestCase):
+    test_plan_name = 'plan name'
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        user_should_have_perm(cls.tester, perm='testplans.change_testplan')
+        cls.product = ProductFactory(name='TestProduct')
+        cls.product_version = VersionFactory(value='0.1', product=cls.product)
+        cls.test_plan_type = PlanTypeFactory()
+        cls.test_plan_1 = TestPlanFactory()
+        cls.testplan_edit_url = reverse('plan-edit', args=[cls.test_plan_1.pk])
+
+        cls.testplan_edit_data = {
+            'author': cls.tester.pk,
+            'product': cls.product.pk,
+            'product_version': cls.product_version.pk,
+            'type': cls.test_plan_type.pk,
+            'name': cls.test_plan_1.name,
+
+            'email_settings-0-auto_to_plan_author': 'off',
+            'email_settings-0-auto_to_case_owner': 'off',
+            'email_settings-0-auto_to_case_default_tester': 'off',
+            'email_settings-0-notify_on_case_update': 'off',
+            'email_settings-0-notify_on_plan_update': 'off',
+
+            'email_settings-0-id': cls.test_plan_1.emailing.pk,
+            'email_settings-TOTAL_FORMS': '1',
+            'email_settings-INITIAL_FORMS': '1',
+            'email_settings-MIN_NUM_FORMS': '0',
+            'email_settings-MAX_NUM_FORMS': '1',
+            'is_active': True,
+        }
+
+    def test_show_testplan_edit_page(self):
+        response = self.client.get(self.testplan_edit_url)
+        self.assertContains(response, _('Edit TestPlan'))
+        self.assert_notify_form(response)
+        self.assertContains(
+            response,
+            '<input class="bootstrap-switch" name="is_active" type="checkbox" checked>',
+            html=True)
+
+    def test_edit_testplan_text_field(self):
+        edit_data = self.testplan_edit_data.copy()
+        new_text = 'Edited: {0}'.format(self.test_plan_1.text)
+        edit_data['text'] = new_text
+
+        response = self.client.post(self.testplan_edit_url, data=edit_data, follow=True)
+
+        self.assertContains(response, new_text, html=True)
+        self.test_plan_1.refresh_from_db()
+        self.assertEqual(new_text, self.test_plan_1.text)
+
+    def test_try_editting_testplan_with_invalid_notify_form(self):
+        test_plan = TestPlanFactory(
+            author=self.tester)
+
+        testplan_edit_url = reverse('plan-edit', args=[test_plan.pk])
+        edit_data = self.testplan_edit_data.copy()
+        edit_data['email_settings-0-auto_to_plan_author'] = 'invalid_value'
+        edit_data['is_active'] = False
+
+        response = self.client.post(testplan_edit_url, data=edit_data, follow=True)
+        self.assert_notify_form(response)
```

Comment:
I'm not really sure what's the expected behavior here. I can't reproduce it manually but if one of these ON/OFF buttons is invalid I would expect an error and not the default value to be present. 

But I'm not entirely certain how the code underneath behaves. Could be a bug there.

Reply:
I see your comment here:

> Case 2 - When there is an invalid value on notify form:
> notify_formset comes with the invalid value on rendered_content after render_to_response it returns back to the  default value, where all the notify values are on(checked).

so I'm inclined to mark this as resolved. But you really don't need is_active here.

Reply:
I tried explicitly modifying the underlying field for notify_form inside the view in order to make it invalid. The cleaned result still came out as True!.

This is probably some quirk in Django internals which I don't have more time to spend figuring out. Looks like these boolean fields do not fail on invalid values but tolerate them. 

Please add a comment in this test scenario to indicate that this is the case and that's why all of the buttons are still ON. And this scenario is done.